src/ape/api/transactions.py

```diff
@@ -231,10 +233,23 @@ def decode_logs(self, abi: Union[EventABI, "ContractEvent"]) -> Iterator[Contrac
         Returns:
             Iterator[:class:`~ape.types.ContractLog`]
         """
-        if not isinstance(abi, EventABI):
-            abi = abi.abi
+        if abi:
+            if not isinstance(abi, EventABI):
+                abi = abi.abi
 
-        yield from self.provider.network.ecosystem.decode_logs(abi, self.logs)
+            yield from self.provider.network.ecosystem.decode_logs(abi, self.logs)
+        else:
+            # if abi is not provided, decode all events
+            for log in self.logs:
+                contract_type = self.chain_manager.contracts.get(log["address"])
+                try:
+                    event_abi = contract_type.events[log["topics"][0]]  # type: ignore
```

Comment:
any alternative to using `# type ignore` here?

maybe:

```
selector = str(log["topics"][0])
event_abi = contract_type.events[selector]
```

Reply:
i think we could, but the type should be bytes or HexBytes